src/core/src/bound_evaluate.cpp

```diff
@@ -80,8 +80,17 @@ ov::Tensor evaluate_bound(const Output<Node>& output, bool is_upper, bool invali
             ov::TensorVector outputs;
             for (const auto& out : node->outputs()) {
                 const auto& out_shape = out.get_partial_shape();
-                auto shape = out_shape.is_static() ? out_shape.to_shape() : Shape(out_shape.rank().get_length());
-                outputs.emplace_back(out.get_element_type(), shape);
+                const auto& out_et = out.get_element_type();
+
```

Comment:
Can we have test for your changes please?

Reply:
Based on reproduction scenario add tests to simulate similar situation when outputs can have dynamic element type or dynamic rank.